hw/ip/otbn/dv/otbnsim/sim/insn.py

```diff
@@ -896,6 +902,11 @@ def execute(self, state: OTBNState) -> None:
         addr = (grs1_val + self.offset) & ((1 << 32) - 1)
 
         grs2_val = state.gprs.get_reg(self.grs2).read_unsigned()
+
+        if grs2_val > 0x1f:
+            state.stop_at_end_of_cycle(ILLEGAL_INSN)
+            return
```

Comment:
This code works simulation wise (from some brief testing) but may not be great from a documentation psuedo-code perspective. Perhaps we'd prefer

```python
grs1_val = state.gprs.get_reg(self.grs1).read_unsigned()
addr = (grs1_val + self.offset) & ((1 << 32) - 1)

grs2_val = state.gprs.get_reg(self.grs2).read_unsigned()

if grs2_val <= 0x1f:
    wrs = grs2_val & 0x1f
    wrs_val = state.wdrs.get_reg(wrs).read_unsigned()
    
    state.dmem.store_u256(addr, wrs_val)
    
    if self.grs1_inc:
        new_grs1_val = (grs1_val + 32) & ((1 << 32) - 1)
        state.gprs.get_reg(self.grs1).write_unsigned(new_grs1_val)
    
    if self.grs2_inc:
        new_grs2_val = grs2_val + 1
        state.gprs.get_reg(self.grs2).write_unsigned(new_grs2_val)
else:
    raise_error(ILLEGAL_INSN)
```

Reply:
I'm happy to take a look, can do that as a follow-up PR so we can discuss appropriate substitutions there unless we're particularly keen to resolve this in this PR.